frontends/p4/evaluator/evaluator.h

```diff
@@ -67,6 +67,9 @@ class Evaluator final : public Inspector, public IHasBlock {
     { setValue(expression, expression); return false; }
     bool preorder(const IR::BoolLiteral* expression) override
     { setValue(expression, expression); return false; }
+    bool preorder(const IR::ListLiteral* expression) override
+    { setValue(expression, expression); return false; }
```

Comment:
I'm sure you have a good reason, but do you mind explaining why do we need to re-evaluate an ListLiteral? A comment will help, especially since we are just setting expression to expression.

Reply:
Actually the point her is to NOT reevaluate the ListLiteral -- ListLiteral is a subclass of ListExpression, so if we did not define this preorder function (which does nothing, just adds the literal to the map as itself), the preorder(ListExpression) function would be called to create a bew ListLiteral.